File: Exercises/rb101-rb109_small_problems/easy_7/multiply_lists.rb
# frozen_string_literal: true

def multiply_list(numbers1, numbers2)
  numbers1.map { |num| num * numbers2[numbers1.index(num)] }
end

puts multiply_list([3, 5, 7], [9, 10, 11]) == [27, 50, 77]

def multiply_list_alt1(numbers1, numbers2)
  numbers1.zip(numbers2).map { |numbers| numbers.reduce(:*) }
end

puts multiply_list_alt1([3, 5, 7], [9, 10, 11]) == [27, 50, 77]

File: js210/exercises/easy_4/multiply_lists.js
function multiplyList(array1, array2) {
  let products = [];

  for (let index = 0; index < array1.length; index += 1) {
    products.push(array1[index] * array2[index]);
  }

  return products;
}

console.log(multiplyList([3, 5, 7], [9, 10, 11]));    // [27, 50, 77]
